src/lib/viewers/BaseViewer.js

```diff
@@ -270,6 +274,23 @@ class BaseViewer extends EventEmitter {
         this.destroyed = true;
     }
 
+    /**
+     * Handles a download error when using a non default host.
+     *
+     * @param {Error} err - Load error
+     * @param {string} downloadURL - download URL
+     * @return {void}
+     */
+    handleDownloadError(err, downloadURL) {
+        if (isNonDefaultDownloadHost(downloadURL)) {
+            setDownloadHostFallback();
+            this.load();
```

Comment:
This still has a few problems as some of our UI methods called in load are not item potent, investigating further.